template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -243,7 +250,39 @@ private Map<String, Set<ServiceComponent>> getNonGatewayServicesByHostGroup() {
                                 .filter(i -> !i.getComponent().equalsIgnoreCase("GATEWAY"))
                                 .collect(toSet())
                 ));
+    }
+
+    private Set<String> collectComponentByHostGroupWithRoleDependencies() {
+        Map<String, Set<ServiceComponent>> hgToNonGwServiceComponents = getNonGatewayServicesByHostGroup();
+        return hgToNonGwServiceComponents.values().stream()
+                .filter(serviceComponents -> !Strings.isNullOrEmpty(getDependentService(serviceComponents)))
```

Comment:
We cannot filter out a component if dependencies are not known.

ROLE_DEPENDENCIES is not exhaustive. If a dependency is not defined for a component residing on the hostGroup we want to scale - we have to assume that we don't know the dependencies, and return something along the lines of "dependencies_cannot_be_determined"

Example: A HG containing 'NodeManager', 'DataNode' and 'RegionServer". We must be able to determine all dependent host groups. The current code will ignore DN and RS, and only return the hostgroup(s) for the ResourceManager. If we attempt to scale and the NameNode is non-functional - the operation will fail.